crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1377,6 +1376,38 @@ impl<'src> Parser<'src> {
             }
         }
 
+        if has_tstring {
+            match tstring_count.cmp(&strings.len()) {
```

Comment:
I'd prefer using `tstring_count < strings.len()` over a match with an `unreachable`. The idea of using a `match` is to handle all cases but this doesn't seem very useful if one of the cases is impossible anyway